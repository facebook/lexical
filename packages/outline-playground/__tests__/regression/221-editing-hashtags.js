/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {
  initializeE2E,
  assertHTML,
  assertSelection,
  focusEditor,
  waitForSelector,
} from '../utils';

describe('Regression test #221', () => {
  initializeE2E((e2e) => {
    it(`Can handle space in hashtag`, async () => {
      const {page} = e2e;

      await focusEditor(page);
      await page.keyboard.type('#yolo');
      await waitForSelector(page, '.editor-text-hashtag');
      await assertHTML(
        page,
        '<p class="editor-paragraph PlaygroundEditorTheme__ltr gkum2dnh" dir="ltr"><span class="editor-text-hashtag" data-outline-text="true">#yolo</span></p>',
      );
      await assertSelection(page, {
        anchorPath: [0, 0, 0],
        anchorOffset: 5,
        focusPath: [0, 0, 0],
        focusOffset: 5,
      });

      await page.keyboard.press('ArrowLeft');
      await page.keyboard.press('ArrowLeft');
      await page.keyboard.press('Space');
      await assertHTML(
        page,
        '<p class="editor-paragraph PlaygroundEditorTheme__ltr gkum2dnh" dir="ltr"><span class="editor-text-hashtag" data-outline-text="true">#yo</span><span data-outline-text="true"> lo</span></p>',
      );
      await assertSelection(page, {
        anchorPath: [0, 1, 0],
        anchorOffset: 1,
        focusPath: [0, 1, 0],
        focusOffset: 1,
      });
    });

    it(`Can handle delete in hashtag`, async () => {
      const {page} = e2e;

      await focusEditor(page);
      await page.keyboard.type('#yolo ');
      await waitForSelector(page, '.editor-text-hashtag');
      await assertHTML(
        page,
        '<p class="editor-paragraph PlaygroundEditorTheme__ltr gkum2dnh" dir="ltr"><span class="editor-text-hashtag" data-outline-text="true">#yolo</span><span data-outline-text="true"> </span></p>',
      );
      await assertSelection(page, {
        anchorPath: [0, 1, 0],
        anchorOffset: 1,
        focusPath: [0, 1, 0],
        focusOffset: 1,
      });

      await page.keyboard.press('ArrowLeft');
      await page.keyboard.press('Delete');
      await assertHTML(
        page,
        '<p class="editor-paragraph PlaygroundEditorTheme__ltr gkum2dnh" dir="ltr"><span class="editor-text-hashtag" data-outline-text="true">#yolo</span></p>',
      );
      await assertSelection(page, {
        anchorPath: [0, 0, 0],
        anchorOffset: 5,
        focusPath: [0, 0, 0],
        focusOffset: 5,
      });
    });

    it(`Can handle backspace into hashtag`, async () => {
      const {page} = e2e;

      await focusEditor(page);
      await page.keyboard.type('#yolo ');
      await waitForSelector(page, '.editor-text-hashtag');
      await assertHTML(
        page,
        '<p class="editor-paragraph PlaygroundEditorTheme__ltr gkum2dnh" dir="ltr"><span class="editor-text-hashtag" data-outline-text="true">#yolo</span><span data-outline-text="true"> </span></p>',
      );
      await assertSelection(page, {
        anchorPath: [0, 1, 0],
        anchorOffset: 1,
        focusPath: [0, 1, 0],
        focusOffset: 1,
      });

      await page.keyboard.press('Backspace');
      await page.keyboard.press('Backspace');
      await assertHTML(
        page,
        '<p class="editor-paragraph PlaygroundEditorTheme__ltr gkum2dnh" dir="ltr"><span class="editor-text-hashtag" data-outline-text="true">#yol</span></p>',
      );
      await assertSelection(page, {
        anchorPath: [0, 0, 0],
        anchorOffset: 4,
        focusPath: [0, 0, 0],
        focusOffset: 4,
      });
    });
  });
});
