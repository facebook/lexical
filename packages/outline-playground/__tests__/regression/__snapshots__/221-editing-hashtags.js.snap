// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression test #221 Can handle backspace into hashtag 1`] = `
<p
  class="editor-paragraph"
  dir="ltr"
>
  <span
    class="editor-text-hashtag"
  >
    #yolo
  </span>
  <span>
     
  </span>
</p>
`;

exports[`Regression test #221 Can handle backspace into hashtag 2`] = `
<p
  class="editor-paragraph"
  dir="ltr"
>
  <span
    class="editor-text-hashtag"
  >
    #yol
  </span>
</p>
`;

exports[`Regression test #221 Can handle delete in hashtag 1`] = `
<p
  class="editor-paragraph"
  dir="ltr"
>
  <span
    class="editor-text-hashtag"
  >
    #yolo
  </span>
  <span>
     
  </span>
</p>
`;

exports[`Regression test #221 Can handle delete in hashtag 2`] = `
<p
  class="editor-paragraph"
  dir="ltr"
>
  <span
    class="editor-text-hashtag"
  >
    #yolo
  </span>
</p>
`;

exports[`Regression test #221 Can handle space in hashtag 1`] = `
<p
  class="editor-paragraph"
  dir="ltr"
>
  <span
    class="editor-text-hashtag"
  >
    #yolo
  </span>
</p>
`;

exports[`Regression test #221 Can handle space in hashtag 2`] = `
<p
  class="editor-paragraph"
  dir="ltr"
>
  <span
    class="editor-text-hashtag"
  >
    #yo
  </span>
  <span>
     lo
  </span>
</p>
`;
